foreman/foreman.spec

```diff
@@ -540,10 +544,24 @@ cp %%{buildroot}%%{%{name}_bundlerd_dir}/%%{gem_name}.rb ./bundler.d/%%{gem_name
 unlink tmp \\
 \\
 export BUNDLER_EXT_NOSTRICT=1 \\
-/usr/bin/%%{?scl:%%{scl}-}rake %%{-r*}%%{!?-r:plugin:assets:precompile[%%{-n*}%%{!?-n:%%{gem_name}}]} RAILS_ENV=production --trace \\
+%%{?-s:/usr/bin/%%{?scl:%%{scl}-}rake %%{-r*}%%{!?-r:plugin:assets:precompile[%%{-n*}%%{!?-n:%%{gem_name}}]} RAILS_ENV=production --trace} \\
+%%{?-a:/usr/bin/%%{?scl:%%{scl}-}rake plugin:apipie:cache[%%{gem_name}] RAILS_ENV=development cache_part=resources OUT=%%{buildroot}%%{gem_instdir}/public/apipie-cache/plugin/%%{gem_name} #gem_instdir}/public/apipie-cache/plugin/%%{gem_name} --trace} \\
 \\
 popd \\
-rm -rf ./usr
+rm -rf ./usr \\
+%%{?-a:mkdir -p %%{buildroot}%%{foreman_dir}/public/apipie-cache/plugin} \\
+%%{?-a:ln -s %%{gem_instdir}/public/apipie-cache/plugin/%%{gem_name} %%{buildroot}%%{foreman_dir}/public/apipie-cache/plugin/%%{gem_name}}
+EOF
+
+cat > %{buildroot}%{_sysconfdir}/rpm/macros.%{name}-apipie << EOF
+# Common locations
+%%%{name}_apipie_cache_plugin %%{gem_instdir}/public/apipie-cache/plugin/%%{gem_name}
+%%%{name}_apipie_cache_foreman %%{foreman_dir}/public/apipie-cache/plugin/%%{gem_name}
+
+%%%{name}_apipie_cache_plugin_post \\
++cp -r %%{foreman_apipie_cache_foreman}/* %%{foreman_dir}/public/apipie-cache/ \\
```

Comment:
For a plugin there is one copy of the prebuilt cache in `%{gem_instdir}/public/apipie-cache/plugin/%{gem_name}` and it is linked to the `%{foreman_dir}/public/apipie-cache/plugin/%%{gem_name}`.
The plugin can contailn the same resources as the foreman core (e.g. environments in Foreman and Katello) so it is not possible to link it to `%{foreman_dir}/public/apipie-cache/` directly in build time as it would cause conflicts. Therefore we copy it in post and overwriting the duplicites.

I agree it would be better to have just one copy in the package. And I would prefere to drop the copy in the `gem_instdir`. Would that be okay?